added a section to search news and events using metadata.

dadrasarmin opened the pull request:
I find it sad that we cannot search news and events using metadata that we provide for each post. Now, it is a demonstration that we can search posts using title, dates, tags, supporters, subsites, and authors.

Of course, it can improve but I think having something to do some queries is nice. It is also asked in different issues before such as these:

- https://github.com/galaxyproject/galaxy-hub/issues/407
- https://github.com/galaxyproject/galaxy-hub/issues/1099
- https://github.com/galaxyproject/galaxy-hub/issues/2200
- https://github.com/galaxyproject/galaxy-hub/issues/2864
- https://github.com/galaxyproject/galaxy-hub/issues/3135
- https://github.com/galaxyproject/galaxy-hub/issues/1537
- https://github.com/galaxyproject/galaxy-hub/issues/252

We do not have to include it anywhere in the NavBar. I find its current place in the NavBar ugly. But I wanted to check whether there is any interest in such a thing and then I can improve the aesthetic.

arash77 commented on src/components/NavBar.vue:
```diff
@@ -23,6 +23,8 @@
                 </b-navbar-nav>
                 <b-navbar-nav id="global-items" class="ml-auto">
+                    <!-- our new search-metadata link -->
+                    <b-nav-item to="/Search2" exact-active-class="active"> Meta-Search </b-nav-item>
```
I think this should help the lint error:
```suggestion
                    <b-nav-item to="/search2" exact-active-class="active"> Meta-Search </b-nav-item>
```

dadrasarmin commented:
The current lint error is not related to what I suggested in this PR and is mentioned previously in this issue: https://github.com/galaxyproject/galaxy-hub/issues/387 and it is about an internal page that does not exist now but there is a link to it here: https://github.com/galaxyproject/galaxy-hub/blob/e3eba062c97d2af0a0a3d058be60ebe8ad6d529d/content/learn/advanced-workflow/basic-editing/index.md?plain=1#L124

bgruening commented:
@dannon are you fine with it? It will not be visible for now, so we can try it in production and enable it in the main pages if it turn out to be useful.

dannon commented:
@bgruening I've been wanting to take a look at this but have not had time yet with GCC coming up.  I am not opposed to improving it of course, but what has to be kept in mind is that this is a 100% prebuilt static site -- there are no real dynamic queries here.  So any query is done clientside against some dataset, which is something prebuilt and shipped over the wire.  I would want to see how this affected the payload, how large an index we expect at 1000 posts, etc.

That said I don't mind at all this being (perhaps temporarily) in a second 'advanced news and events search' type page, but at the top level I am hesitant to add an extra search to the site -- I think there is significant value in having a single search that *just works* regardless of what you put in it.

dadrasarmin commented:
Thank you for your positive and honest feedback. I spent some time trying to reduce the size of the index and refactor the code, but haven’t landed on a sustainable solution that fully addresses the concerns raised. Because this PR doesn’t yet solve the core issues, I’m going to close it for now.

I’ve enjoyed working on this and appreciate how supportive this community has been. I look forward to contributing again in the future. Thank you.

dannon commented:
Hey, I just wanted to follow up and say thanks for jumping in and trying to build this out and enhance things!  It was a cool idea, but the static site performance stuff is just one of those annoying constraints we have to deal with.  Either way I'm glad you enjoyed working on this!
